### ISSUE TITLE
Retained output grad requirement is always True

### ISSUE THREAD

**Issue by hvy** [chainer/chainer#5422]
`requires_grad` is always `True` for retrieved retained outputs. 
Is it not more correct to set it to `False` unless double backprop is enabled?
https://github.com/chainer/chainer/blob/82f5d896017bfab35a541c5a5c7efda744641d69/chainer/function_node.py#L646

**Comment by toslunar** [comment]
If the method `get_retained_outputs` of `FunctionNode` is called, the variable created in the line is required by the function node.

**Comment by hvy** [comment]
I see. My question was if it's correct to return the outputs as they were during forward, in `get_retained_outputs`. For instance, if the backward implementation simply returns the retained outputs, the creator node of the computed gradient is not `None` even when `enable_double_backprop` is `False`.

**Comment by toslunar** [comment]
I don't see your question is about reconstructed retained outputs or about all retained outputs.

**Comment by hvy** [comment]
I was referring to all retentions (considering the reconstruction as an implementation detail).

**Comment by hvy** [comment]
Bump. This ticket is kept alive by https://github.com/chainer/chainer/issues/6116.

**Comment by toslunar** [comment]
I'd rather not increase the difference of the behavior between the cases the output variable is alive or not.  For example, it may depend on iterations whether the GC runs.

**Comment by hvy** [comment]
Although stale, let me reopen it as it's not finished.
